overlord/hookstate/ctlcmd/set.go

```diff
@@ -79,6 +79,11 @@ func (s *setCommand) Execute(args []string) error {
 
 		transaction.Set(s.context().SnapName(), key, value)
 	}
+	context.Lock()
+	if context.IsSnapContext() {
+		transaction.Commit()
```

Comment:
this looks suspicious, we should instead call context.Done() I think from some place higher even in this case